OpenRA.Mods.Common/Traits/Render/RenderDetectionCircle.cs

```diff
@@ -45,8 +66,36 @@ public IEnumerable<IRenderable> RenderAfterWorld(WorldRenderer wr)
 				self.CenterPosition,
 				range,
 				0,
-				Color.FromArgb(128, Color.LimeGreen),
-				Color.FromArgb(96, Color.Black));
+				info.Color,
+				info.ContrastColor);
+
+			if (info.DrawUpdateLine)
+			{
+				for (var i = info.LineTrailLength; i >= 0; i--)
+				{
+					var angle = new WAngle((lineAngle - i * info.UpdateLineTick) % 1024);
+					var length = new WVec((int)(range.Length * (angle.Cos() / 1024f)), (int)(range.Length * (angle.Sin() / 1024f)), 0);
```

Comment:
You should be able to simplify this to `var length = range.Length * new WVec(angle.Cos(), angle.Sin(), 0) / 1024;`